Client/examples/client_lib_basic_example.cpp

```diff
@@ -0,0 +1,131 @@
+#include <ClientConfiguration.h>
+#include <chronolog_client.h>
+
+#include <chrono_monitor.h>
+#include <cmd_arg_parse.h>
+
+#include <cassert>
+#include <common.h>
+#include <unistd.h>
+
+int main(int argc, char** argv)
+{
+
+    // Load configuration
+    std::string conf_file_path = parse_conf_path_arg(argc, argv);
+    chronolog::ClientConfiguration confManager;
+    if(!conf_file_path.empty())
+    {
+        if(!confManager.load_from_file(conf_file_path))
+        {
+            std::cerr << "[ClientExample] Failed to load configuration file '" << conf_file_path
+                      << "'. Using default values instead." << std::endl;
+        }
+        else
+        {
+            std::cout << "[ClientExample] Configuration file loaded successfully from '" << conf_file_path << "'."
+                      << std::endl;
+        }
+    }
+    else
+    {
+        std::cout << "[ClientExample] No configuration file provided. Using "
+                     "default values."
+                  << std::endl;
+    }
+    confManager.log_configuration(std::cout);
+
+    // Initialize logging
+    int result = chronolog::chrono_monitor::initialize(
+            confManager.LOG_CONF.LOGTYPE, confManager.LOG_CONF.LOGFILE, confManager.LOG_CONF.LOGLEVEL,
+            confManager.LOG_CONF.LOGNAME, confManager.LOG_CONF.LOGFILESIZE, confManager.LOG_CONF.LOGFILENUM,
+            confManager.LOG_CONF.FLUSHLEVEL);
+    if(result == 1) { return EXIT_FAILURE; }
+
+    // Build portal configs
+    chronolog::ClientPortalServiceConf portalConf;
+    portalConf.PROTO_CONF = confManager.PORTAL_CONF.PROTO_CONF;
+    portalConf.IP = confManager.PORTAL_CONF.IP;
+    portalConf.PORT = confManager.PORTAL_CONF.PORT;
+    portalConf.PROVIDER_ID = confManager.PORTAL_CONF.PROVIDER_ID;
+
+    // Build query configs
+    chronolog::ClientQueryServiceConf queryConf;
+    queryConf.PROTO_CONF = confManager.QUERY_CONF.PROTO_CONF;
+    queryConf.IP = confManager.QUERY_CONF.IP;
+    queryConf.PORT = confManager.QUERY_CONF.PORT;
+    queryConf.PROVIDER_ID = confManager.QUERY_CONF.PROVIDER_ID;
+
+    LOG_INFO("[ClientExample] Starting ChronoLog Client Example");
+
+    // Create a ChronoLog client
+    chronolog::Client client(portalConf, queryConf);
+
+    // Connect to ChronoVisor
+    int ret = client.Connect();
+    std::cout << "[ClientExample] Connect returned: " << ret << "\n";
```

Comment:
is it better to call `chronolog::to_string_client(ret)` to have a more readable output?